### ISSUE TITLE
subrepo clone "doesn't contain upstream HEAD" on some tags

### ISSUE THREAD

**Issue by mmerickel** [ingydotnet/git-subrepo#109]
First a clone using release tag 2.5.3:

```
git subrepo clone https://github.com/AFNetworking/AFNetworking.git foo -b 2.5.3 -v -d
* Fetch the upstream: https://github.com/AFNetworking/AFNetworking.git (2.5.3).
  * Fetch 'https://github.com/AFNetworking/AFNetworking.git' (2.5.3).
>>> git fetch --quiet https://github.com/AFNetworking/AFNetworking.git 2.5.3
  * Get the upstream subrepo HEAD commit.
>>> git rev-parse FETCH_HEAD
  * Create ref 'refs/subrepo/foo/fetch'.
* Make the directory 'foo/' for the clone.
>>> mkdir -p foo
* Commit the new 'foo/' content.
  * Check that '598eec5e1a1248f1bb313e5b7fcf9b6dfed00e01' exists.
  * Make sure '598eec5e1a1248f1bb313e5b7fcf9b6dfed00e01' contains the upstream HEAD.
  * Put remote subrepo content into 'foo/'.
>>> git read-tree --prefix=foo -u 598eec5e1a1248f1bb313e5b7fcf9b6dfed00e01
  * Put info into 'foo/.gitrepo' file.
>>> git config --file=foo/.gitrepo subrepo.remote https://github.com/AFNetworking/AFNetworking.git
>>> git config --file=foo/.gitrepo subrepo.branch 2.5.3
>>> git config --file=foo/.gitrepo subrepo.commit 598eec5e1a1248f1bb313e5b7fcf9b6dfed00e01
>>> git config --file=foo/.gitrepo subrepo.parent 0a22598cad6b6507544747fc99e1291f61503892
>>> git config --file=foo/.gitrepo subrepo.cmdver 0.3.0
>>> git add foo/.gitrepo
  * Commit to the 'foo' branch.
>>> git commit -m git subrepo clone --branch=2.5.3 https://github.com/AFNetworking/AFNetworking.git foo subrepo: subdir: "foo" merged: "598eec5" upstream: origin: "https://github.com/AFNetworking/AFNetworking.git" branch: "2.5.3" commit: "598eec5" git-subrepo: version: "0.3.0" origin: "https://github.com/ingydotnet/git-subrepo.git" commit: "c0fc373"
  * Create ref 'refs/subrepo/foo/commit'.
Subrepo 'https://github.com/AFNetworking/AFNetworking.git' (2.5.3) cloned into 'foo'.
```

Second a clone using the 2.6.3 release tag:

```
git subrepo clone https://github.com/AFNetworking/AFNetworking.git foo -b 2.6.3 -v -d
* Fetch the upstream: https://github.com/AFNetworking/AFNetworking.git (2.6.3).
  * Fetch 'https://github.com/AFNetworking/AFNetworking.git' (2.6.3).
>>> git fetch --quiet https://github.com/AFNetworking/AFNetworking.git 2.6.3
  * Get the upstream subrepo HEAD commit.
>>> git rev-parse FETCH_HEAD
  * Create ref 'refs/subrepo/foo/fetch'.
* Make the directory 'foo/' for the clone.
>>> mkdir -p foo
* Commit the new 'foo/' content.
  * Check that 'afae578b883ec1d2222007af3b2abdb67d474eca' exists.
  * Make sure 'afae578b883ec1d2222007af3b2abdb67d474eca' contains the upstream HEAD.
git-subrepo: Can't commit: 'afae578b883ec1d2222007af3b2abdb67d474eca' doesn't contain upstream HEAD
```

I'm unable to understand what the difference is between these two tags that is causing things to fail on 2.6.3. Can someone enlighten me? I'm trying to use git subrepo to vendor libraries into our main project, and update them periodically to newer release tags.

**Comment by grimmySwe** [comment]
It's seems that 2.6.3 is actually an annotated tag, 2.5.3 is a lightweight one. So when we perform the fetch our FETCH_HEAD will in the 2.6.3 case contain the tag hash, not the commit hash. So to when we try to use that hash later it will fail...

Workaround for now is to use the actual commit hash for the tag 2.6.3 = ee89cc988e2709a025908ef952a7377ce7ba83b8, then it will work. But I will check if this can actually be solved in some other better way.

**Comment by grimmySwe** [comment]
I created a fix on issue/107 branch so feel free to check it out and verify it.

**Comment by mmerickel** [comment]
Confirmed this fixed my issue *if* I reclone the subrepo. I had previously had 2.5.3 cloned using 0.2.1 and attempted to do a `git subrepo pull foo -b 2.6.3` and encountered these errors. If I re-clone 2.5.3 I get a different ``parent`` in the .gitrepo than I had with 0.2.1. When I do a pull it works now however. I don't imagine you intended to make this work bw-compat but wanted to point it out.

**Comment by ingydotnet** [comment]
Fixed by #110
